shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -2125,8 +2141,9 @@ impl KafkaSinkCluster {
                     })
                     .unwrap();
 
-                find_coordinator.host = shotover_node.address.host.clone();
-                find_coordinator.port = shotover_node.address.port;
+                find_coordinator.host =
+                    StrBytes::from_string(shotover_node.address.ip().to_string());
```

Comment:
Converting from SocketAddr (entirely on the stack) to StrBytes (reference counted string stored on the heap) results in an allocation, which would be good to avoid.

How about we have 2 address fields in ShotoverNode.
```rust
    /// Use this field when replacing fields in kafka messages.
    pub address_as_string KafkaAddress,
    /// Use this field when connecting to the shotover node
    pub address_as_socket_addr: SocketAddr,
```
Since ShotoverNode is treated as immutable we dont have to worry about "source of truth" issues with this duplication.